packages/core/src/logger.js

```diff
@@ -5,17 +5,43 @@
 
 'use strict';
 
+/**
+ * @type {*}
+ */
```

Comment:
My take on that is this: something of type any should be in the vast majority of times replaced by a proper type.
However, there will be cases where a variable can actually be anything, which can pretty much be true in our project. eg: a function that copies an object.

In the few cases where I left a variable typed as `*` was either because it really fits as anything, or it was something that I was unsure on how to type it properly, at least so far.

But as a rule of thumb, something typed as `*` tells me something. Either it is REALLY anything given a specific context, or that I should review it and check if there is anything I can do to type it properly.
One more thing that I assumed for this first type checking work was to allow myself to be slightly flexible, because 1) I am still getting familiar with the code; 2) this is just one directory in the whole project; and 3) some better polished typing may require some code change, which I was not willing to risk at this stage :)